### PR TITLE
update main and app

### PR DESCRIPTION
**Pull request by mingsorawit**
- change js file to tsx file
- change css
- create page structure

design URL: https://zpl.io/bePZpvz

28/6
-change form.tsx as a page into MailchimpForm.tsx as a component
-import MailchimpForm.tsx and use it in main
-update ui components in MailchimpForm.tsx
-npm install @fortawesome/fontawesome-svg-core @fortawesome/free-solid-svg-icons @fortawesome/react-fontawesome
for icon that will be used in submit button
Screenshot (1120)
Screenshot (1121)
Screenshot (1122)

format files
disabled input and button when sending
Screenshot (1133)
Screenshot (1127)

made it responsive when width is less than 650px
minor change in MailchimpForm.module.css (margin, etc.)
Screenshot (1134)
Screenshot (1135)

### CODE REVIEW

**Comment by mingsorawit** [comment]
install npm i react-mailchimp-subscribe
create basic form that connects with beamdata mailchimp

File: styles/Home.module.css [hunk 1]
```diff
@@ -45,9 +47,21 @@
 }
 
 .title {
-  margin: 0;
-  line-height: 1.15;
-  font-size: 4rem;
+  
+  /*width: 631px;*/
```
Comment (oukunanan) [inline styles/Home.module.css:51]: Remove

File: styles/Home.module.css [hunk 2]
```diff
@@ -45,9 +47,21 @@
...
+  height: 40px;
+  margin-bottom: 36px;
+  font-family: 'Lexend Deca', sans-serif;
+  font-size: 32px;
+  /*font-weight: normal;
+  font-stretch: normal;
+  font-style: normal;*/
```
Comment (oukunanan) [inline styles/Home.module.css:58]: Remove

File: pages/form.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    email &&
+    email.indexOf("@") > -1 &&
+    onValidated({
+      EMAIL: email,
+    });
+  }
+
+  console.log(email);
+  console.log(status);
+  console.log(message);
```
Comment (oukunanan) [inline pages/form.tsx:32]: Remove consoe.log

File: pages/form.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  message: string | Error | null;
+  onValidated: (obj: {EMAIL: string;}) => void;
+};
+
+const CustomForm = ({ status, message, onValidated }: Props) => {
+  
+  const [email, setEmail] = useState('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if(event.target){
```
Comment (oukunanan) [inline pages/form.tsx:15]: Why we need this `if` here?

File: pages/form.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      setEmail(event.target.value);
+    }
+  }
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    alert('An email was submitted') 
+    event.preventDefault();
+    email &&
+    email.indexOf("@") > -1 &&
```
Comment (oukunanan) [inline pages/form.tsx:24]: Why we need this?

File: pages/index.tsx [hunk 6]
```diff
@@ -1,27 +1,37 @@
...
-        <link rel="icon" href="/favicon.ico" />
+        <title>Beam Data Early Access</title>
+        {/* <meta name="description" content="Generated by create next app" />
+        <link rel="icon" href="/favicon.ico" /> */}
+        <link rel="preconnect" href="https://fonts.googleapis.com"></link>
+        <link rel="preconnect" href="https://fonts.gstatic.com"></link>
+        <link href="https://fonts.googleapis.com/css2?family=Lexend+Deca&display=swap" rel="stylesheet"></link>
       </Head>
 
+      <div className={styles.header}>Beam</div>
```
Comment (oukunanan) [inline pages/index.tsx:19]: I thougt P'Sandee already import the logo

File: pages/form.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
+import React, { useState }from "react";
+import MailchimpSubscribe from "react-mailchimp-subscribe";
+
+type Props = {
+  status: string | null;
+  message: string | Error | null;
+  onValidated: (obj: {EMAIL: string;}) => void;
+};
+
+const CustomForm = ({ status, message, onValidated }: Props) => {
```
[the remark this answers is not in the document]
Comment (oukunanan) [reply pages/form.tsx:10]: And prefer normal function than arrow function
https://kentcdodds.com/blog/function-forms

File: package.json [hunk 8]
```diff
@@ -12,13 +12,16 @@
     "format:write": "prettier --write ."
   },
   "dependencies": {
+    "dotenv": "^10.0.0",
```
Comment (oukunanan) [inline package.json:15]: I think we can config `next.config.js` instead of install this package.

File: pages/form.tsx [hunk 9]
```diff
@@ -0,0 +1,66 @@
+import React, { useState }from "react";
```
Comment (oukunanan) [inline pages/form.tsx:1]: So we gonna have url `/form` ? according to your file name

File: pages/form.tsx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  const [email, setEmail] = useState('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if(event.target){
+      setEmail(event.target.value);
+    }
+  }
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    alert('An email was submitted')
```
Comment (oukunanan) [inline pages/form.tsx:21]: Remove

File: pages/form.tsx [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+      <div className="mc__field-container">
+        <input type="email" value={email} placeholder="your@email.com" onChange={(event) => handleChange(event)} required/>
+      </div>
+      <input type='submit' value="Submit"/>
+    </form>
+  );
+};
+
+
+const MailchimpFormContainer = () => {
```
Comment (oukunanan) [inline pages/form.tsx:46]: Convert to normal function

File: pages/index.tsx [hunk 12]
```diff
@@ -1,27 +1,37 @@
...
+
       <Head>
-        <title>Create Next App</title>
-        <meta name="description" content="Generated by create next app" />
-        <link rel="icon" href="/favicon.ico" />
+        <title>Beam Data Early Access</title>
+        {/* <meta name="description" content="Generated by create next app" />
+        <link rel="icon" href="/favicon.ico" /> */}
+        <link rel="preconnect" href="https://fonts.googleapis.com"></link>
+        <link rel="preconnect" href="https://fonts.gstatic.com"></link>
```
Comment (oukunanan) [inline pages/index.tsx:15]: Save font to `public` folder and use font-face to import the font in global.css

File: pages/index.tsx [hunk 13]
```diff
@@ -1,27 +1,37 @@
...
 
-        <div className={styles.grid}>
+        <div>
+          <form>
+            <input type="email" placeholder="Your work email address" className={styles.input}/>
+            <button className={styles.button}>Get Access</button>
+          </form>
+        </div>
+
+        {/* <div className={styles.grid}>
```
Comment (oukunanan) [inline pages/index.tsx:34]: Please remove unused code entire project.

File: styles/Home.module.css [hunk 14]
```diff
@@ -45,9 +47,21 @@
...
+  line-height: 1.75;
+  letter-spacing: normal;
```
Comment (oukunanan) [inline styles/Home.module.css:60]: Why?

File: pages/form.tsx [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+      EMAIL: email,
+    });
+  }
+
+  console.log(email);
+  console.log(status);
+  console.log(message);
+
+  return (
+    <form className="mc__form" onSubmit={(event) => handleSubmit(event)}>
```
Comment (oukunanan) [inline pages/form.tsx:35]:
```suggestion
    <form className="mc__form" onSubmit={handleSubmit}>
```

File: pages/form.tsx [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+      <h3 className="mc__title">Get early access for Enterprise</h3>      
+      <div className="mc__field-container">
+        <input type="email" value={email} placeholder="your@email.com" onChange={(event) => handleChange(event)} required/>
```
Comment (oukunanan) [inline pages/form.tsx:38]:
```suggestion
        <input type="email" value={email} placeholder="your@email.com" onChange={handleChange} required/>
```

File: pages/form.tsx [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+  );
+};
+
+
+const MailchimpFormContainer = () => {
+
+  const postUrl = `https://beamdata.us18.list-manage.com/subscribe/post?u=${process.env.NEXT_PUBLIC_MAILCHIMP_U}&id=${process.env.NEXT_PUBLIC_MAILCHIMP_ID}`;
+  
+  return (
+    <div className="mc__form-container">
```
Comment (oukunanan) [inline pages/form.tsx:51]: Where these className come from?

File: pages/form.tsx [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+      <MailchimpSubscribe 
+        url={postUrl}
+        render={({ subscribe, status, message }) => (
+          <CustomForm
+            status={status}
+            message={message}
+            onValidated={formData => subscribe(formData)}
```
Comment (oukunanan) [inline pages/form.tsx:58]: Can we do this?
```suggestion
            onValidated={subscribe}
```

**Review by oukunanan — CHANGES_REQUESTED** [review]

**Comment by oukunanan** [comment]
Please also put design url within PR description.

**Comment by mingsorawit** [comment]
update code as suggested
display message when submit form depends on status
status could be success/error/sending
Screenshot (1116)
Screenshot (1117)

**Review by oukunanan — APPROVED** [review]
Approved but will do refactor in the future PR.

**Pull request merged**